pkg/pool-linear/contracts/erc4626/ERC4626LinearPool.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ERC20.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+
+import "../interfaces/IERC4626.sol";
+
+import "../LinearPool.sol";
+
+contract ERC4626LinearPool is LinearPool {
+    IERC4626 private immutable _wrappedToken;
+
+    constructor(
+        IVault vault,
+        string memory name,
+        string memory symbol,
+        IERC20 mainToken,
+        IERC20 wrappedToken,
+        uint256 upperTarget,
+        uint256 swapFeePercentage,
+        uint256 pauseWindowDuration,
+        uint256 bufferPeriodDuration,
+        address owner
+    )
+        LinearPool(
+            vault,
+            name,
+            symbol,
+            mainToken,
+            wrappedToken,
+            upperTarget,
+            swapFeePercentage,
+            pauseWindowDuration,
+            bufferPeriodDuration,
+            owner
+        )
+    {
+        _require(address(mainToken) == IERC4626(address(wrappedToken)).underlying(), Errors.TOKENS_MISMATCH);
+        _wrappedToken = IERC4626(address(wrappedToken));
+    }
+
+    function _getWrappedTokenRate() internal view override returns (uint256) {
+        // Rate between wrapped and their underlying token at _wrappedToken.decimals() scale
+        uint256 rate = _wrappedToken.assetsPerShare();
+
+        uint256 tokenDecimals = ERC20(address(_wrappedToken)).decimals();
+        uint256 decimalsDifference = Math.sub(18, tokenDecimals);
+
+        // Upscale to e18
+        return rate * 10 ** decimalsDifference;
```

Comment:
Made a comment (now deleted) here before that we can just use the scaling factor from `LinearPool` but that includes a rate can't be used.

We can store `10 ** decimalsDifference` in an immutable variable to avoid having to make many calls out to `_wrappedToken` however.

We can also make `getWrappedToken()` in `LinearPool` public to avoid having to store `_wrappedToken` again in this contract.

Reply:
@mosinnik would you add https://github.com/balancer-labs/balancer-v2-monorepo/commit/4939eafab6d10bf55df9af7c3c15ca581bc01931 to this branch re: storing scaling in an immutable var?

@TomAFrench - I wasn't sure if I should modify `LinearPool` if it's in the wild already.  Should I add the helper method to `LinearPool` or this derived contract?

Reply:
It's fine to modify LinearPool in this way. We won't be changing the interface by making such a change.

Reply:
:+1:  done https://github.com/balancer-labs/balancer-v2-monorepo/commit/85aa152fdd09bc205a017cabc07da15eeee0bfc5